src/Traits/MutatesMigrationCommands.php

```diff
@@ -51,12 +53,14 @@ public function handle()
         $this->input->setOption('force', true);
         $this->input->setOption('database', $this->connection->tenantName());
 
-        $this->processHandle(function ($website) {
+        $this->processHandle(function (Website $website, Collection $websites) {
```

Comment:
Instead of passing the full collection (thus creating a full copy in memory), why don't we make it a boolean?

Reply:
@fletch3555 I think making it a boolean would make `processHandle` method too much concentrated to this issue. Maybe passing `chunkSize` integer would be a better idea? What do you think?

Reply:
I mean `processHandle` method is not responsible for managing the connection and it should know nothing about it, so adding connection specific logic seems not right to me. All it does is running the provided callback on the chunks of tenants. So passing the information about the chunk seems more clean to me:

```php
$this->processHandle(function (Website $website, $chunkSize) {
    $this->connection->set($website);

    parent::handle();

    if ($chunkSize > 1) {
        $this->connection->purge();
    }
});
```

Reply:
Fletch is right though. You're sending an argument into a method that will possibly increase the memory footprint. Either we move the purge out of the processHandle or we use a boolean argument (bool $purge = true).

What do you think?

Reply:
Hey @luceos and @fletch3555, are there still any problems with this PR?